### ISSUE TITLE
Require (and potentially enforce) attribute-style `// issue:12345` comments in issue-related tests

### ISSUE THREAD

**Issue by tgross35** [rust-lang/rust#113604]
This comes as a followup to #113583

The goal is to add meta comments to all tests related to an issue or ICE (e.g. regression tests) so that we can programmatically extract this information. For example:

```rust
// issue:12345
// issue:6789
// ice:1234

// sample link to different repo:
// issue: rust-lang/cargo:1234   
```

We should make having this a goal for all regression tests going forward, and as a requirement with any renaming that might happen to suit #113583. We can likely do some of this programatically.

**Comment by workingjubilee** [comment]
We should probably allow naming issues that aren't in the main repo, so it should probably be more like `{repo}#{issue}`.

**Comment by matthiaskrgr** [comment]
Yeah we need a clear naming scheme for subtrees, for example when you add a test to clippy, inside the clippy repo, `issue:123` looks fine because of course it refers to the clippy repo. But then it gets subtree-merged into the rustc repo and would point at some random rust-lang/rust ticket from 7 years ago (because the test is somewhere in  src/tools/clippy inside the rustc repo now?)
